libs/dscache/odc/dscache/apps/dstiler.py

```diff
@@ -36,6 +40,17 @@ def cli(native, native_albers, web, dbfile):
         gs = web_gs(web)
         group_key_fmt = 'web_' + str(web) + '/{:03d}_{:03d}'
         binner = lambda dss: bin_dataset_stream(gs, dss)
+    elif crs is not None or resolution is not None or shape is not None or bin_format is not None:
+        settings = dict(crs=crs, resolution=resolution, shape=shape, bin_format=bin_format)
+        missing = [key for key, value in settings.items() if value is None]
+        if missing:
+            raise ValueError('Missing options in custom grid spec: {}'.format(' '.join(missing)))
+        group_key_fmt = bin_format
+        resolution = [float(comp) for comp in resolution.split(',')]
+        shape = [float(comp) for comp in shape.split(',')]
+        tile_size = [abs(res * shp) for res, shp in zip(resolution, shape)]
+        gs = GridSpec(crs=geom.CRS(crs), tile_size=tile_size, resolution=resolution)
```

Comment:
most things now accept crs as a string on input, if `GridSpec` doesn't we should fix that in core